app/models/miq_expression.rb

```diff
@@ -740,17 +740,8 @@ def _to_sql(exp, tz)
         ids = klass.find_tagged_with(:any => tag, :ns => ns).pluck(:id)
         clause = klass.send(:sanitize_sql_for_conditions, ["#{klass.table_name}.id IN (?)", ids])
       else
-        db, field = exp[operator]["field"].split(".")
-        model = db.constantize
-        assoc, field = field.split("-")
-        ref = model.reflect_on_association(assoc.to_sym)
-
-        inner_where = "#{field} = '#{exp[operator]["value"]}'"
-        if cond = ref.scope          # Include ref.options[:conditions] in inner select if exists
-          cond = extract_where_values(ref.klass, cond)
-          inner_where = "(#{inner_where}) AND (#{cond})"
-        end
-        clause = "#{model.table_name}.id IN (SELECT DISTINCT #{ref.foreign_key} FROM #{ref.table_name} WHERE #{inner_where})"
+        field = Field.parse(exp[operator]["field"])
```

Comment:
I hadn't realized that `MiqExpression::Field` is in darga. wow @imtayadeway you were working on that refactoring for a long time (I knew it was a lot of work, but didn't remember just how much)